src/main/scala/zio/prelude/Inverse.scala

```diff
@@ -62,10 +62,801 @@ object Inverse extends Lawful[EqualInverse] {
+    A: Inverse,
+    B: Inverse,
+    C: Inverse,
+    D: Inverse,
+    E: Inverse,
+    F: Inverse,
+    G: Inverse,
+    H: Inverse,
+    I: Inverse,
+    J: Inverse,
+    K: Inverse,
+    L: Inverse,
+    M: Inverse,
+    N: Inverse,
+    O: Inverse,
+    P: Inverse,
+    Q: Inverse,
+    R: Inverse,
+    S: Inverse,
+    T: Inverse,
+    U: Inverse
+  ]: Inverse[(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U)] =
+    makeFrom(
+      Identity.Tuple21Identity, {
+        case (
+            (a1, b1, c1, d1, e1, f1, g1, h1, i1, j1, k1, l1, m1, n1, o1, p1, q1, r1, s1, t1, u1),
+            (a2, b2, c2, d2, e2, f2, g2, h2, i2, j2, k2, l2, m2, n2, o2, p2, q2, r2, s2, t2, u2)
+            ) =>
+          (
+            a1 ~~ a2,
+            b1 ~~ b2,
+            c1 ~~ c2,
+            d1 ~~ d2,
+            e1 ~~ e2,
+            f1 ~~ f2,
+            g1 ~~ g2,
+            h1 ~~ h2,
+            i1 ~~ i2,
+            j1 ~~ j2,
+            k1 ~~ k2,
+            l1 ~~ l2,
+            m1 ~~ m2,
+            n1 ~~ n2,
+            o1 ~~ o2,
+            p1 ~~ p2,
+            q1 ~~ q2,
+            r1 ~~ r2,
+            s1 ~~ s2,
+            t1 ~~ t2,
+            u1 ~~ u2
+          )
+      }
+    )
+
+  /**
+   * Derives an `Inverse` for a product type given an `Inverse` for
+   * each element of the product type.
+   */
+  implicit def Tuple22Inverse[
+    A: Inverse,
+    B: Inverse,
+    C: Inverse,
+    D: Inverse,
+    E: Inverse,
+    F: Inverse,
+    G: Inverse,
+    H: Inverse,
+    I: Inverse,
+    J: Inverse,
+    K: Inverse,
+    L: Inverse,
+    M: Inverse,
+    N: Inverse,
+    O: Inverse,
+    P: Inverse,
+    Q: Inverse,
+    R: Inverse,
+    S: Inverse,
+    T: Inverse,
+    U: Inverse,
+    V: Inverse
+  ]: Inverse[(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V)] =
+    makeFrom(
+      Identity.Tuple22Identity, {
+        case (
+            (a1, b1, c1, d1, e1, f1, g1, h1, i1, j1, k1, l1, m1, n1, o1, p1, q1, r1, s1, t1, u1, v1),
+            (a2, b2, c2, d2, e2, f2, g2, h2, i2, j2, k2, l2, m2, n2, o2, p2, q2, r2, s2, t2, u2, v2)
+            ) =>
+          (
+            a1 ~~ a2,
+            b1 ~~ b2,
+            c1 ~~ c2,
+            d1 ~~ d2,
+            e1 ~~ e2,
+            f1 ~~ f2,
+            g1 ~~ g2,
+            h1 ~~ h2,
+            i1 ~~ i2,
+            j1 ~~ j2,
+            k1 ~~ k2,
+            l1 ~~ l2,
+            m1 ~~ m2,
+            n1 ~~ n2,
+            o1 ~~ o2,
+            p1 ~~ p2,
+            q1 ~~ q2,
+            r1 ~~ r2,
+            s1 ~~ s2,
+            t1 ~~ t2,
+            u1 ~~ u2,
+            v1 ~~ v2
+          )
+      }
+    )
+
+}
+
+trait InverseSyntax {
+
+  /**
+   * Provides infix syntax for combining two values with an inverse
+   * operation.
+   */
+  implicit class InverseOps[A](l: A) {
+
+    /**
+     * A symbolic alias for `inverse`.
+     */
+    def ~~(r: => A)(implicit inverse: Inverse[A]): A =
```

Comment:
I've picked `~~` as the symbolic alias for `inverse`. But feel free to suggest a better name, I was empty on ideas :smile:

Reply:
Yeah, it is definitely a decent one. Another idea would be `-` since it isn't overloaded by `anystring2add` but it probably has more risks of being shadowed by another definition and is not really consistent with `<>`. `~<>` is perhaps the most accurate but a little long. I don't really have a better suggestion either which is why I merged as is but always think it is good to be careful with new symbolic operators, especially when it is not an associative operation. 😃